src/Compilers/CSharp/Portable/Symbols/UnboundGenericType.cs

```diff
@@ -119,4 +119,74 @@ public override int GetHashCode()
                 : Hash.Combine(_name, _errorInfo.Code);
         }
     }
+
+    /// <summary>
+    /// Used for lightweight binding of type constraints. Instead of binding type arguments,
+    /// we'll just use these placeholders instead. That's good enough binding to compute
+    /// <see cref="TypeSymbol.IsValueType"/> with minimal binding.
+    /// </summary>
+    internal sealed class PlaceholderTypeArgumentSymbol : ErrorTypeSymbol
+    {
+        private static readonly TypeWithAnnotations s_instance = TypeWithAnnotations.Create(new PlaceholderTypeArgumentSymbol());
+
+        public static ImmutableArray<TypeWithAnnotations> CreateTypeArguments(ImmutableArray<TypeParameterSymbol> typeParameters)
+        {
+            var result = ArrayBuilder<TypeWithAnnotations>.GetInstance();
+            foreach (var typeParameter in typeParameters)
+            {
+                result.Add(s_instance);
+            }
+            return result.ToImmutableAndFree();
+        }
+
+        private PlaceholderTypeArgumentSymbol(TupleExtraData? tupleData = null)
```

Comment:
>TupleExtraData? tupleData = null [](start = 46, length = 32)

If `tupleData` can be non-null, `Equals()` needs to be updated to check reference equality only.